source/common/upstream/upstream_impl.cc

```diff
@@ -758,7 +727,8 @@ ClusterInfoImpl::ClusterInfoImpl(
                                   ? std::make_unique<OptionalClusterStats>(
                                         config, *stats_scope_, factory_context.clusterManager())
                                   : nullptr),
-      features_(parseFeatures(config, http_protocol_options_)),
+      features_(ClusterInfoImpl::HttpProtocolOptionsConfigImpl::parseFeatures(
+          config, http_protocol_options_)),
```

Comment:
It seems odd to me that we are calling into extension code still in the upstream code? But I guess that is unavoidable due to legacy design here? (Ultimately it would be nice if the core cluster code had no real understanding of HTTP but that is probably a bit too asipirational for now.)